index.js

```diff
@@ -33,25 +35,29 @@ module.exports = function(options) {
       }
       const newType = 'image/jpeg';
       return new Promise((resolve, reject) => {
-        require('child_process').exec(se([ options.tifig, file.path, newPath ]), { encoding: 'utf8' }, function(error, stdout, stderr) {
-          if (error) {
-            console.log(stdout);
-            console.error(stderr);
-            return reject(error);
-          }
-          file.name = newName;
-          // Avoid leaking many megabytes of disk space
-          require('fs').unlinkSync(file.path);
-          file.path = newPath;
-          file.type = newType;
-          return resolve(true);
+        const errorHandler = (error) => {
+            console.error(error);
+            reject(error);
+        };
+        worker.once('message', (_message) => {
```

Comment:
ohh this is receiving a message from the worker. OK.

Reply:
Yup. Both the parent and child process communicate to each other by emitting and subscribing to the `message` event. Also be noted that in this case we register the listener for a single invocation so we don't trigger more than one callback at a time due to concurrent promises. Would you like some more comments on the code?